src/standard_wrappers/nft_standards/dip_721_v2.ts

```diff
@@ -0,0 +1,108 @@
+import { Actor, ActorSubclass, HttpAgent } from '@dfinity/agent';
+import { Principal } from '@dfinity/principal';
+
+import { NFTDetails } from '../../interfaces/nft';
+import Interface, { TokenMetadata, GenericValue } from '../../interfaces/dip_721_v2';
+import IDL from '../../idls/dip_721_v2.did';
+import NFT from './default';
+import { NFT as NFTStandard} from '../../constants/standards';
+import { ok } from 'assert';
+
+interface Property {
+  name: string;
+  value: string;
+}
+
+interface MetadataKeyVal { 'key' : string, 'val' : GenericValue }
+
+interface Metadata {
+  [key: string]: { value: MetadataKeyVal , purpose: string } | Array<Property> | string;
+  properties: Array<Property>;
+}
+
+const extractMetadataValue = (metadata: any) => {
+  const metadataKey = Object.keys(metadata)[0];
```

Comment:
Can we guarantee that we'll always have an item on the list? Otherwise, maybe wrap in a try/catch block if we don't want to spend too much time on it.

Reply:
Yeah, the return type from canister will always have a GenericValue item (first key, name chages)
This is a function straight from dabs dip721v1

Reply:
👍